receiver/kubeletstatsreceiver/runnable.go

```diff
@@ -86,7 +91,7 @@ func (r *runnable) Run() error {
 	}
 
 	metadata := kubelet.NewMetadata(r.extraMetadataLabels, podsMetadata)
-	mds := kubelet.MetricsData(r.logger, summary, metadata, typeStr, r.metricGroupsToCollect)
+	mds := kubelet.MetricsData(r.logger, summary, metadata, typeStr, r.metricGroupsToCollect, r.getPersistentVolumeLabelsFromClaim)
```

Comment:
Can't we add conditionally prefetched `volumeMetadata` to the `metadata` structure above instead of passing around `volumeClaimLabelsSetter` ?

Reply:
I guess we could that instead. It also seems like doing that might make it cleaner to cache information about volumes. So, the `Metadata` struct will gather all the required metadata about volumes upfront and later on map it back to the corresponding volumes while actually collecting the metrics. Will try that out and push an update.

Reply:
Right now, as errors are encountered while collecting metadata about volumes, they're logged and metric collection for that specific volume is skipped. If the receiver were to collect all the metadata upfront, to achieve the current behavior, it would also need to keep track of the errors along with the collected metadata so we can log out the exact error. So, I moved the volumes metadata setter method to the `Metadata` struct for now but it's still invoked when the metrics are collected. Let me know what you think.